### PR TITLE
[inventec] Replace os.system and remove subprocess with shell=True

### PR DESCRIPTION
**Pull request by maipbui**
Signed-off-by: maipbui <maibui@microsoft.com>

Dependency: [https://github.com/sonic-net/sonic-buildimage/pull/12065](https://github.com/sonic-net/sonic-buildimage/pull/12065)
#### Why I did it
1. `getstatusoutput` is used without a static string and it uses `shell=True`
2. `subprocess()` - when using with `shell=True` is dangerous. Using subprocess function without a static string can lead to command injection.
3. `os` - not secure against maliciously constructed input and dangerous if used to evaluate dynamic content.
#### How I did it
1. use `getstatusoutput` without shell=True
2. `subprocess()` - use `shell=False` instead. use an array string. Ref: [https://semgrep.dev/docs/cheat-sheets/python-command-injection/#mitigation](https://semgrep.dev/docs/cheat-sheets/python-command-injection/#mitigation)
3. `os` - use with `subprocess`
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

File: platform/broadcom/sonic-platform-modules-inventec/common/utils/asic_monitor.py [hunk 1]
```diff
@@ -66,7 +64,7 @@ class BCMUtil(bcmshell):
     
     def get_platform(self):
         if self.platform is None:
-            self.platform = os.popen("uname -n").read().strip()
```
Comment (qiluo-msft) [inline platform/broadcom/sonic-platform-modules-inventec/common/utils/asic_monitor.py:69]: >.strip() [](http://example.com/codeflow?start=55&length=8)

Do you want to keep `.strip()`? It also applies to blank char.

File: platform/broadcom/sonic-platform-modules-inventec/d6332/utils/inventec_d6332_util.py [hunk 2]
```diff
@@ -236,8 +237,9 @@ def system_install(boot_option):
 	      return status
     for addr_offset in range (0,FAN_NUM):
         addr=FAN_VPD_ADDR_BASE+addr_offset
-        cmd = "i2cdetect -y "+str(FAN_VPD_CHANNEL)+" "+str(addr)+" "+str(addr)+" | grep "+str(hex(addr)).replace('0x','')
-        result=os.system(cmd)
+        cmd1 = ["i2cdetect", "-y", str(FAN_VPD_CHANNEL), str(addr), str(addr)]
+        cmd2 = ["grep", str(hex(addr)).replace('0x','')]
```
Comment (qiluo-msft) [inline platform/broadcom/sonic-platform-modules-inventec/d6332/utils/inventec_d6332_util.py:241]: >replace [](http://example.com/codeflow?start=39&length=7)

better way: [https://stackoverflow.com/a/16414603/2514803](https://stackoverflow.com/a/16414603/2514803)

Comment (maipbui) [reply platform/broadcom/sonic-platform-modules-inventec/d6332/utils/inventec_d6332_util.py]: Fixed, thanks!

**Review by qiluo-msft — APPROVED** [review]

**Comment by maipbui** [comment]
@klhaung @david-xk @CynthiaINV Could you help review and verify?

**Review by linalvin — APPROVED** [review]

**Pull request merged**